src/ripple/app/misc/SHAMapStoreImp.h

```diff
@@ -190,7 +190,7 @@ class SHAMapStoreImp : public SHAMapStore
     {
         std::uint64_t check = 0;
 
-        for (uint256 it: cache.getKeys())
+        for (auto const& it: cache.getKeys())
```

Comment:
Good spotting getting rid of all those copies.  Does it make sense to rename `it` to something like `key`, since it isn't an iterator?

I'm in the minority here, but I prefer the old `uint256` declaration over the `auto` in this particular case.  Since `CacheInstance` is a template parameter, there's no telling what its `getKeys()` will return.  However (at the moment) `database_->fetchNode()` insists that its parameter be a `uint256`.  Since this is not really generic code, it's easier for me to read if I know the type.  But I don't expect you to change this, since I'm in the minority.